src/test/java/io/jenkins/plugins/folderauth/FolderAuthorizationWebAPITest.java

```diff
@@ -202,13 +206,145 @@ public void testDeleteAgentRole() throws IOException {
+        assertEquals("Verifying that role is extracted correctly",
+                new HashSet(Arrays.asList("folder1", "folder2")),
+                new HashSet((List) json.get("folders")));
+
+        // Testing getting a role that doesn't exists
+        methodURL = new URL(apiURL + "/getFolderRole?name=nonexistent");
+        request = new WebRequest(methodURL, HttpMethod.GET);
+        page = webClient.getPage(request);
+        assertEquals("Testing if request is successful", HttpURLConnection.HTTP_OK, page.getWebResponse().getStatusCode());
+        roleString = page.getWebResponse().getContentAsString();
+        assertEquals("{}", roleString);  // expecting an empty response as the role doesn't exists.
+    }
+
+    @Test
+    public void testGetGlobalRole() throws IOException {
+        testAssignSidToGlobalRole();  // creates a folder role named "folderRole" and assigns "alice" to the role
+
+        String roleName = "globalRole";
+        URL methodURL = new URL(apiURL + "/getGlobalRole?name=" + roleName);
+        WebRequest request = new WebRequest(methodURL, HttpMethod.GET);
+
+        Page page = webClient.getPage(request);
+        // Verifying that web request is successful and that the role is found
+        assertEquals("Testing if request is successful", HttpURLConnection.HTTP_OK, page.getWebResponse().getStatusCode());
+        String roleString = page.getWebResponse().getContentAsString();
+        JSONObject json = JSONObject.fromObject(roleString);
+        assertEquals(roleName, json.get("name"));
+        JSONArray sids = (JSONArray) json.get("sids");
+        assertTrue("User alice should be assigned to role", sids.contains("alice"));
+
+        // Testing getting a role that doesn't exists
+        methodURL = new URL(apiURL + "/getGlobalRole?name=nonexistent");
+        request = new WebRequest(methodURL, HttpMethod.GET);
+        page = webClient.getPage(request);
+        assertEquals("Testing if request is successful", HttpURLConnection.HTTP_OK, page.getWebResponse().getStatusCode());
+        roleString = page.getWebResponse().getContentAsString();
+        assertEquals("{}", roleString);  // expecting an empty response as the role doesn't exists.
+    }
+
+    @Test
+    public void testGetAssignedRoles() throws IOException {
+        testAssignSidToAgentRole();
+        testAssignSidToFolderRole();
+        testAssignSidToGlobalRole();
+        // above creates 3 roles and assigns "alice" to all 3
+
+        URL methodURL = new URL(apiURL + "/getAssignedRoles?sid=alice");
+        WebRequest request = new WebRequest(methodURL, HttpMethod.GET);
+        Page page = webClient.getPage(request);
+        assertEquals("Testing if request is successful", HttpURLConnection.HTTP_OK, page.getWebResponse().getStatusCode());
+
+        // Verifying that the method returns all 3 roles
+        String response = page.getWebResponse().getContentAsString();
+        JSONObject json = JSONObject.fromObject(response);
```

Comment:
Is it possible to convert the `JSONObject` back to a `FolderRole` or `AbstractRole` object using http://json-lib.sourceforge.net/apidocs/jdk15/net/sf/json/JSONObject.html#toBean(net.sf.json.JSONObject,%20java.lang.Class) ?

Reply:
Similar to my comment before, the roles need to extend map/bean to work. It'll require a refactor of the existing class